libs-haskell/bazel-runfiles/src/DA/Bazel/Runfiles.hs

```diff
@@ -5,71 +5,45 @@
 -- but since we also want support for locating runfiles in JARs
 -- it is simpler to have all code located here.
 module DA.Bazel.Runfiles
-  ( locateRunfiles
-  , locateRunfilesMb
+  ( create
+  , rlocation
+  , locateRunfiles
   , mainWorkspace
   ) where
 
-import Control.Monad.Trans.Maybe
-import Data.Foldable
-import Data.List
-import Data.List.Split (splitOn)
+import qualified Bazel.Runfiles
 import System.Directory
 import System.Environment
 import System.FilePath
 
 mainWorkspace :: String
 mainWorkspace = "com_github_digital_asset_daml"
 
-locateRunfiles :: FilePath -> IO FilePath
-locateRunfiles target = do
-    dirOrError <- locateRunfilesMb target
-    case dirOrError of
-        Left e -> error e
-        Right d -> pure d
-
-locateRunfilesMb :: FilePath -> IO (Either String FilePath)
-locateRunfilesMb target = do
-    execPath <- getExecutablePath
-    mbDir <- runMaybeT . asum . map MaybeT $
-        [ do let jarResources = takeDirectory execPath </> "resources"
-             hasJarResources <- doesDirectoryExist jarResources
-             pure $ if hasJarResources
-                 then Just jarResources
-                 else Nothing
-        , do let runfilesDir = execPath <> ".runfiles"
-             hasTarget <- doesPathExist (runfilesDir </> target)
-             hasManifest <- doesFileExist (runfilesDir </> "MANIFEST")
-             if hasTarget
-                then pure $ Just(runfilesDir </> target)
-             else if hasManifest
-                then lookupTargetInManifestFile (runfilesDir </> "MANIFEST") target
-             else pure Nothing
-        , do mbDir <- lookupEnv "RUNFILES_DIR"
-             pure (fmap (</> target) mbDir)
-        ]
-    pure $ maybe (Left $ "Could not locate runfiles for target: " <> target) Right mbDir
-
-lookupTargetInManifestFile :: FilePath -> FilePath -> IO (Maybe FilePath)
-lookupTargetInManifestFile manifestPath target = do
-    manifestFile <- readFile manifestPath
-    let manifest = map lineToTuple (lines manifestFile)
-    let targetNormalised = intercalate "/" (splitOn "\\" (normalise target))
-    pure $ asum [findExact targetNormalised manifest, findDir targetNormalised manifest]
+data Runfiles
+  = Resources FilePath
+  | BazelRunfiles Bazel.Runfiles.Runfiles
+
+create :: IO Runfiles
+create = do
+  execPath <- getExecutablePath
+  let jarResources = takeDirectory execPath </> "resources"
+  hasJarResources <- doesDirectoryExist jarResources
+  if hasJarResources
+      then pure $! Resources jarResources
+      else BazelRunfiles <$> Bazel.Runfiles.create
```

Comment:
`create` is a weird name for a function that doesn’t seem to create anything (based on looking at the source) (but ofc that’s not something addressable as part of this PR).

Reply:
It is indeed. This is following the `bazel-runfiles` API, which in turn is following the [Bazel specification](https://docs.google.com/document/d/e/2PACX-1vSDIrFnFvEYhKsCMdGdD40wZRBX3m3aZ5HhVj4CtHPmiXKDCxioTUbYsDydjKtFDAzER5eg7OjJWs3V/pub).

> Every language's library will have a similar interface:
>
> a Create method that inspects the environment and/or argv[0] to determine the runfiles strategy (manifest-based or directory-based; see below), initializes runfiles handling and returns a Runfiles object

Reply:
TIL, thanks! Makes sense to keep that name in that case.